internal/pkg/deploy/pipeline.go

```diff
@@ -442,41 +445,143 @@ func (s *GitHubV1Source) Owner() (string, error) {
 	return owner, nil
 }
 
+type associatedEnvironment struct {
+	// Name of the environment, must be unique within an application.
+	// This is also the name of the pipeline stage.
+	Name string
+
+	// The region this environment is created in.
+	Region string
+
+	// AppName represents the application name the environment is part of.
+	AppName string
+
+	// AccountID of the account this environment is stored in.
+	AccountID string
+}
+
 // PipelineStage represents configuration for each deployment stage
 // of a workspace. A stage consists of the Config Environment the pipeline
 // is deploying to, the containerized services that will be deployed, and
 // test commands, if the user has opted to add any.
 type PipelineStage struct {
-	*AssociatedEnvironment
-	LocalWorkloads   []string
-	RequiresApproval bool
-	TestCommands     []string
+	*associatedEnvironment
+	localWorkloads   []string
```

Comment:
stupid question ✋🏼  why do we prefer making these fields private and use getter to get the value here?

Reply:
lol it's not stupid! It's the idea behind a data abstraction, we can change how things are represented internally but not change the interface. For example, I originally had:
```go
func (stg *PipelineStage) EnvManagerRoleARN() string {
	return fmt.Sprintf("arn:aws:iam::%s:role/%s-%s-EnvManagerRole", stg.accountID, stg.name, stg.envName)
}
```
but then I swapped it to return the `config.Env`'s env manager role ARN.

TBH it doesn't matter much for `PipelineStage`.  
However, I think this approach is nice for `WorkloadDeployAction`. Eventually, we'll have another type `EnvDeployAction` where I want to create a common interface between the two types:
```go
type DeployAction interface { // implemented by both WorkloadDeployAction and EnvDeployAction.
   Name() string
   StackName() string
   RunOrder() int
   TemplatePath() string
   TemplateConfigPath() string
}
```